src/core-concepts-in-embedded-application/logging.md

```diff
@@ -0,0 +1,118 @@
+# Logging
+
+​Logging is a crucial aspect of embedded systems development, providing visibility into the system's behavior and aiding in debugging and monitoring. In the Rust ecosystem for ESP devices, two prominent logging frameworks are commonly used: the [log] crate and [defmt]. The [esp-println] crate serves as a bridge, facilitating logging through these frameworks on ESP devices.​
+
+## The `log` crate
+
+The [log] crate is a widely adopted logging facade in the Rust community. It defines a set of macros (`info!`, `warn!`, `error!`, etc.) to capture log messages at various levels. The actual logging behavior is determined by the selected backend implementing the `Log` trait.
+
+### Integrating `log` with `esp-println`
+
+To utilize the `log` crate with `esp-println`, follow these steps:​
+
+1. Add Dependencies to your `Cargo.toml`:
+```toml
+esp-hal          = { version = "1.0.0-beta.0", features = ["esp32c6"] }
+esp-println      = { version = "0.13.0", features = ["esp32c6", "log"] }
+log              = { version = "0.4.21" }
+```
+Ensure you specify the feature corresponding to your target device (e.g. `esp32c6` for ESP32-C6).
+
+2. Set log level in your `.config.toml`:
+```toml
+[env]
+ESP_LOG="info" # Set desired log level
+```
+
+3. Initialize the Logger:
```

Comment:
This feels very redundant to me. 3 and 4 could be a single code block, with an `// Set up esp-println as the logger` comment added.